src/pyggdrasil/tree_inference/_simulate.py

```diff
@@ -251,8 +251,142 @@ def attach_cells_to_tree(
     Returns:
         binary matrix of shape ``(n_cells, n_sites)``,
           where ``n_sites`` is determined from the ``tree``
+          NOTE: does not trunicate the last row as shown in the SCITE paper
     """
     if n_cells < 1:
         raise ValueError(f"Number of sampled cells {n_cells} cannot be less than 1.")
 
-    raise NotImplementedError("This function needs to be implemented.")
+    # get no of nodes
+    n_nodes = tree.shape[0]
+
+    # sample cell attachment vector
+    sigma = sample_cell_attachment(rng, n_cells, n_nodes, strategy)
+
+    # get ancestor matrix from adjacency matrix
+    ## get shortest path matrix
+    sp_matrix = floyd_warshall(tree)
+    ## converts shortes path to ancestor matrix
+    ancestor_matrix = shortest_path_to_ancestry_matrix(sp_matrix)
+
+    # get mutation matrix
+    mutation_matrix = built_perfect_mutation_matrix(n_nodes, ancestor_matrix, sigma)
+
+    # raise NotImplementedError("This function needs to be implemented.")
+
+    return mutation_matrix
+
+
+def sample_cell_attachment(
+    rng: interface.JAXRandomKey,
+    n_cells: int,
+    n_nodes: int,
+    strategy: CellAttachmentStrategy,
+) -> interface.Cell_Attachment_Vector:
+    """Samples the node attachment for each cell given a uniform prior.
+
+    Args:
+        rng: JAX random key
+        n_cells: number of cells
+        n_nodes: number of nodes including root,
+            nodes counted from 1, root = n_nodes-1
```

Comment:
I did adopt the convention that the root is the highest label node in the tree, i.e. last row/column in the adjacency matrix / last row in the mutation matrix.

Reply:
Isn't the root the 0th node according to the cell attachment vector? If you prefer the latter convention it's still fine, as long it's documented (perhaps in the module docstring?  It's a convention which shouldn't change within this file).